lib/src/components/NxFilterInput/NxFilterInput.tsx

```diff
@@ -29,18 +30,19 @@ const NxFilterInput = forwardRef<HTMLDivElement, Props>(
 
       return (
         <div {...otherProps} className={classes} ref={ref}>
-          <span className="nx-filter-input__add-on"
-                onClick={value && onClear || undefined}>
-            <NxFontAwesomeIcon icon={icon} />
-          </span>
-          <input type="text"
-                 autoComplete="off"
-                 id={inputId || undefined}
-                 value={value}
-                 onChange={inputOnChange}
-                 placeholder={placeholder || undefined}
-                 className="nx-filter-text-input"
-                 disabled={disabled || undefined} />
+          <div className="nx-text-input__box">
+            <span className="nx-filter-input__add-on" onClick={value && onClear || undefined}>
```

Comment:
Is there any purpose to this span, or could the icon be placed directly into the `.nx-text-input__box`?

Reply:
> Is there any purpose to this span, or could the icon be placed directly into the `.nx-text-input__box`?

The icon doesn't support `onClick`.